File: eolis-api/seed.py
import sys, os
sys.path.insert(0, os.path.dirname(__file__))

from app.database import SessionLocal, engine
from app.models import Base, User, Ticket, Message, SatisfactionRating, Notification, FAQ, Log
from app.security import hash_password
from datetime import datetime

def seed():
    Base.metadata.create_all(bind=engine)
    db = SessionLocal()

    print("[1/4] Clearing existing data...")
    from sqlalchemy import text
    db.execute(text(
        "TRUNCATE TABLE otp_codes, password_resets, logs, notifications, satisfaction_ratings, attachments, messages, tickets, faqs, users RESTART IDENTITY CASCADE"
    ))
    db.commit()

    print("[2/4] Creating users...")
    admin = User(first_name="Christian", last_name="Denmeko", username="Christian.DENMEKO",
                 email="admin@eoliscameroun.com", phone="+237690000001",
                 password_hash=hash_password("Admin@2026!"), role="SYSTEM_ADMIN", status="ACTIVE", language="fr")

    ops = User(first_name="Debora", last_name="Denmeko", username="Debora.DENMEKO",
               email="ops@eoliscameroun.com", phone="+237690000002",
               password_hash=hash_password("Ops@2026!"), role="OPS_ADMIN", status="ACTIVE", language="fr")

    agent1 = User(first_name="Jean", last_name="Mbarga", username="Jean.MBARGA",
                  email="agent1@eoliscameroun.com", phone="+237690000003",
                  password_hash=hash_password("Agent@2026!"), role="AGENT", status="ACTIVE", language="fr")

    agent2 = User(first_name="Marie", last_name="Nguema", username="Marie.NGUEMA",
                  email="agent2@eoliscameroun.com", phone="+237690000004",
                  password_hash=hash_password("Agent@2026!"), role="AGENT", status="ACTIVE", language="fr")

    client1 = User(first_name="Thomas", last_name="Kamga", username="Thomas.KAMGA",
                   email="client1@example.com", phone="+237699000001",
                   password_hash=hash_password("Client@2026!"), role="CLIENT", status="ACTIVE", language="fr")

    client2 = User(first_name="Alice", last_name="Fono", username="Alice.FONO",
                   email="client2@example.com", phone="+237699000002",
                   password_hash=hash_password("Client@2026!"), role="CLIENT", status="ACTIVE", language="en")

    pending = User(first_name="Robert", last_name="Etoa", username="Robert.ETOA",
                   email="pending@example.com", phone="+237699000003",
                   password_hash=hash_password("Client@2026!"), role="CLIENT", status="PENDING", language="fr")

    for u in [admin, ops, agent1, agent2, client1, client2, pending]:
        db.add(u)
    db.commit()

    print("[3/4] Creating tickets...")
    t1 = Ticket(ref="REF-2026-0001", client_id=client1.id, agent_id=agent1.id,
                category="Livraison", subcategory="Conteneur bloqué", equipment_type="Conteneur 40 pieds",
                ship_line="MSC", ship_name="MSC Cameroun", voyage_number="CM2026-04", ship_date="2026-04-15",
                description="Mon conteneur 40 pieds est bloqué au port depuis 3 jours sans explication.",
                urgency="HIGH", status="TREATED",
                taken_at=datetime(2026, 4, 16, 9, 0), closed_at=datetime(2026, 4, 16, 14, 30))
    db.add(t1)
    db.commit()

    for content, sender_id, sender_type in [
        ("Bonjour M. Kamga, nous avons bien reçu votre demande. Nous vérifions la situation de votre conteneur.", agent1.id, "AGENT"),
        ("Merci de votre retour rapide. J'attends votre confirmation.", client1.id, "CLIENT"),
        ("Bonne nouvelle ! Votre conteneur a été libéré. Référence de mainlevée : ML-2026-789.", agent1.id, "AGENT"),
    ]:
        db.add(Message(ticket_id=t1.id, sender_id=sender_id, sender_type=sender_type, content=content, is_read=True))
    db.add(SatisfactionRating(ticket_id=t1.id, client_id=client1.id, agent_id=agent1.id, score=5, comment="Service très rapide et efficace !"))
    db.commit()

    t2 = Ticket(ref="REF-2026-0002", client_id=client1.id, agent_id=agent1.id,
                category="Facturation", subcategory="Retard de paiement",
                description="Je n'ai pas encore reçu ma facture pour la livraison du mois dernier.",
                urgency="MEDIUM", status="IN_PROGRESS", taken_at=datetime.utcnow())
    db.add(t2)
    db.commit()
    db.add(Message(ticket_id=t2.id, sender_id=agent1.id, sender_type="AGENT",
                   content="Bonjour, nous vérifions votre dossier de facturation. Pouvez-vous nous fournir votre numéro de commande ?"))
    db.add(Notification(user_id=client1.id, ticket_id=t2.id, type="NEW_MESSAGE",
                        title="Nouveau message", message=f"Nouveau message dans le dossier REF-2026-0002."))
    db.commit()

    t3 = Ticket(ref="REF-2026-0003", client_id=client2.id,
                category="Delivery", subcategory="Delivery delay", equipment_type="20ft Container",
                ship_line="CMA-CGM", ship_name="CMA Douala", voyage_number="DLA2026-07",
                description="My 20ft container delivery is 5 days late. The expected date was April 20th.",
                urgency="HIGH", status="PENDING")
    db.add(t3)
    db.commit()

    t4 = Ticket(ref="REF-2026-0004", client_id=client2.id, agent_id=agent2.id,
                category="Billing", subcategory="Refund",
                description="I was charged twice for my last delivery. I need a refund for the duplicate charge.",
                urgency="LOW", status="TREATED",
                taken_at=datetime(2026, 4, 20, 10, 0), closed_at=datetime(2026, 4, 21, 16, 0))
    db.add(t4)
    db.commit()
    db.add(SatisfactionRating(ticket_id=t4.id, client_id=client2.id, agent_id=agent2.id, score=4, comment="Good service, resolved my issue."))
    db.commit()

    print("[4/4] Creating FAQs...")
    faqs_fr = [
        ("Livraison", "Comment suivre ma livraison ?", "Connectez-vous à votre espace Eolis Connect et consultez vos demandes."),
        ("Livraison", "Que faire si mon conteneur est bloqué au port ?", "Créez une demande dans la catégorie \"Livraison > Conteneur bloqué\"."),
        ("Livraison", "Quels documents faut-il pour récupérer un conteneur ?", "Facture commerciale, connaissement, déclaration en douane, pièce d'identité."),
        ("Facturation", "Comment obtenir ma facture ?", "Vos factures sont disponibles dans votre espace client."),
        ("Facturation", "Comment contester une facture ?", "Créez une demande dans \"Facturation > Paiement incomplet\" ou \"Facturation > Remboursement\"."),
        ("Information", "Quels services propose Eolis Cameroun ?", "Manutention portuaire, transport de conteneurs, fret conventionnel, gestion documentaire."),
        ("Information", "Quels sont les horaires du service client ?", "Lundi au vendredi de 8h00 à 18h00."),
    ]
    for i, (cat, q, a) in enumerate(faqs_fr):
        db.add(FAQ(locale="fr", category=cat, question=q, answer=a, order=i))

    faqs_en = [
        ("Delivery", "How do I track my delivery?", "Log in to your Eolis Connect account and check your requests."),
        ("Delivery", "What should I do if my container is blocked at the port?", "Create a request under \"Delivery > Blocked container\"."),
        ("Delivery", "What documents are needed to pick up a container?", "Commercial invoice, Bill of Lading, customs declaration, and a valid ID."),
        ("Billing", "How do I get my invoice?", "Your invoices are available in your client area."),
        ("Billing", "How do I dispute an invoice?", "Create a request under \"Billing > Incomplete payment\" or \"Billing > Refund\"."),
        ("Information", "What services does Eolis Cameroun offer?", "Port handling, container transport (20ft and 40ft), conventional freight, and document management."),
        ("Information", "What are customer service hours?", "Monday to Friday from 8:00 AM to 6:00 PM."),
    ]
    for i, (cat, q, a) in enumerate(faqs_en):
        db.add(FAQ(locale="en", category=cat, question=q, answer=a, order=i))

    db.commit()
    db.close()

    print("\nDatabase seeded successfully!")
    print("\nTest accounts:")
    print("  System Admin : Christian.DENMEKO / Admin@2026!")
    print("  Ops Admin    : Debora.DENMEKO    / Ops@2026!")
    print("  Agent 1      : Jean.MBARGA       / Agent@2026!")
    print("  Agent 2      : Marie.NGUEMA      / Agent@2026!")
    print("  Client 1     : Thomas.KAMGA      / Client@2026!")
    print("  Client 2 (EN): Alice.FONO        / Client@2026!")
    print("  Pending      : Robert.ETOA       / Client@2026!")

if __name__ == "__main__":
    seed()

File: eolis-connect/prisma/seed.ts
import 'dotenv/config'
import { pbkdf2Sync, randomBytes } from 'node:crypto'
import { PrismaClient } from '../src/generated/prisma/client'
import { PrismaLibSql } from '@prisma/adapter-libsql'
const dbUrl = 'file:dev.db'
const adapter = new PrismaLibSql({ url: dbUrl })
const db = new PrismaClient({ adapter })

function hash(password: string): string {
  const salt = randomBytes(16).toString('hex')
  const h = pbkdf2Sync(password, salt, 100_000, 32, 'sha256').toString('hex')
  return `pbkdf2:${salt}:${h}`
}

async function main() {
  console.log('🌱 Seeding database...')

  await db.log.deleteMany()
  await db.notification.deleteMany()
  await db.satisfactionRating.deleteMany()
  await db.attachment.deleteMany()
  await db.message.deleteMany()
  await db.ticket.deleteMany()
  await db.passwordReset.deleteMany()
  await db.fAQ.deleteMany()
  await db.user.deleteMany()

  const admin = await db.user.create({
    data: {
      firstName: 'Christian',
      lastName: 'Denmeko',
      username: 'Christian.DENMEKO',
      email: 'admin@eoliscameroun.com',
      phone: '+237690000001',
      passwordHash: hash('Admin@2026!'),
      role: 'SYSTEM_ADMIN',
      status: 'ACTIVE',
      language: 'fr',
    },
  })

  const ops = await db.user.create({
    data: {
      firstName: 'Debora',
      lastName: 'Denmeko',
      username: 'Debora.DENMEKO',
      email: 'ops@eoliscameroun.com',
      phone: '+237690000002',
      passwordHash: hash('Ops@2026!'),
      role: 'OPS_ADMIN',
      status: 'ACTIVE',
      language: 'fr',
    },
  })

  const agent1 = await db.user.create({
    data: {
      firstName: 'Jean',
      lastName: 'Mbarga',
      username: 'Jean.MBARGA',
      email: 'agent1@eoliscameroun.com',
      phone: '+237690000003',
      passwordHash: hash('Agent@2026!'),
      role: 'AGENT',
      status: 'ACTIVE',
      language: 'fr',
    },
  })

  const agent2 = await db.user.create({
    data: {
      firstName: 'Marie',
      lastName: 'Nguema',
      username: 'Marie.NGUEMA',
      email: 'agent2@eoliscameroun.com',
      phone: '+237690000004',
      passwordHash: hash('Agent@2026!'),
      role: 'AGENT',
      status: 'ACTIVE',
      language: 'fr',
    },
  })

  const client1 = await db.user.create({
    data: {
      firstName: 'Thomas',
      lastName: 'Kamga',
      username: 'Thomas.KAMGA',
      email: 'client1@example.com',
      phone: '+237699000001',
      passwordHash: hash('Client@2026!'),
      role: 'CLIENT',
      status: 'ACTIVE',
      language: 'fr',
    },
  })

  const client2 = await db.user.create({
    data: {
      firstName: 'Alice',
      lastName: 'Fono',
      username: 'Alice.FONO',
      email: 'client2@example.com',
      phone: '+237699000002',
      passwordHash: hash('Client@2026!'),
      role: 'CLIENT',
      status: 'ACTIVE',
      language: 'en',
    },
  })

  await db.user.create({
    data: {
      firstName: 'Robert',
      lastName: 'Etoa',
      username: 'Robert.ETOA',
      email: 'pending@example.com',
      phone: '+237699000003',
      passwordHash: hash('Client@2026!'),
      role: 'CLIENT',
      status: 'PENDING',
      language: 'fr',
    },
  })

  // Tickets
  const t1 = await db.ticket.create({
    data: {
      ref: 'REF-2026-0001',
      clientId: client1.id,
      agentId: agent1.id,
      category: 'Livraison',
      subcategory: 'Conteneur bloqué',
      equipmentType: 'Conteneur 40 pieds',
      shipLine: 'MSC',
      shipName: 'MSC Cameroun',
      voyageNumber: 'CM2026-04',
      shipDate: '2026-04-15',
      description: 'Mon conteneur 40 pieds est bloqué au port depuis 3 jours sans explication. J\'ai tous mes documents en règle.',
      urgency: 'HIGH',
      status: 'TREATED',
      takenAt: new Date('2026-04-16T09:00:00'),
      closedAt: new Date('2026-04-16T14:30:00'),
    },
  })

  await db.message.create({
    data: {
      ticketId: t1.id,
      senderId: agent1.id,
      senderType: 'AGENT',
      content: 'Bonjour M. Kamga, nous avons bien reçu votre demande. Nous vérifions la situation de votre conteneur avec le port.',
      isRead: true,
      readAt: new Date('2026-04-16T09:30:00'),
    },
  })

  await db.message.create({
    data: {
      ticketId: t1.id,
      senderId: client1.id,
      senderType: 'CLIENT',
      content: 'Merci de votre retour rapide. J\'attends votre confirmation.',
      isRead: true,
      readAt: new Date('2026-04-16T10:00:00'),
    },
  })

  await db.message.create({
    data: {
      ticketId: t1.id,
      senderId: agent1.id,
      senderType: 'AGENT',
      content: 'Bonne nouvelle ! Votre conteneur a été libéré. Vous pouvez le récupérer à partir de demain matin. Référence de mainlevée : ML-2026-789.',
      isRead: true,
      readAt: new Date('2026-04-16T14:00:00'),
    },
  })

  await db.satisfactionRating.create({
    data: {
      ticketId: t1.id,
      clientId: client1.id,
      agentId: agent1.id,
      score: 5,
      comment: 'Service très rapide et efficace ! Merci beaucoup.',
    },
  })

  const t2 = await db.ticket.create({
    data: {
      ref: 'REF-2026-0002',
      clientId: client1.id,
      agentId: agent1.id,
      category: 'Facturation',
      subcategory: 'Retard de paiement',
      description: 'Je n\'ai pas encore reçu ma facture pour la livraison du mois dernier.',
      urgency: 'MEDIUM',
      status: 'IN_PROGRESS',
      takenAt: new Date(),
    },
  })

  await db.message.create({
    data: {
      ticketId: t2.id,
      senderId: agent1.id,
      senderType: 'AGENT',
      content: 'Bonjour, nous sommes en train de vérifier votre dossier de facturation. Pouvez-vous nous fournir votre numéro de commande ?',
      isRead: false,
    },
  })

  await db.notification.create({
    data: {
      userId: client1.id,
      ticketId: t2.id,
      type: 'NEW_MESSAGE',
      title: 'Nouveau message',
      message: `Vous avez un nouveau message concernant votre dossier REF-2026-0002.`,
      isRead: false,
    },
  })

  const t3 = await db.ticket.create({
    data: {
      ref: 'REF-2026-0003',
      clientId: client2.id,
      category: 'Delivery',
      subcategory: 'Delivery delay',
      equipmentType: '20ft Container',
      shipLine: 'CMA-CGM',
      shipName: 'CMA Douala',
      voyageNumber: 'DLA2026-07',
      description: 'My 20ft container delivery is 5 days late. The expected date was April 20th.',
      urgency: 'HIGH',
      status: 'PENDING',
    },
  })

  const t4 = await db.ticket.create({
    data: {
      ref: 'REF-2026-0004',
      clientId: client2.id,
      agentId: agent2.id,
      category: 'Billing',
      subcategory: 'Refund',
      description: 'I was charged twice for my last delivery. I need a refund for the duplicate charge.',
      urgency: 'LOW',
      status: 'TREATED',
      takenAt: new Date('2026-04-20T10:00:00'),
      closedAt: new Date('2026-04-21T16:00:00'),
    },
  })

  await db.satisfactionRating.create({
    data: {
      ticketId: t4.id,
      clientId: client2.id,
      agentId: agent2.id,
      score: 4,
      comment: 'Good service, resolved my issue. Could have been faster.',
    },
  })

  // FAQs FR
  const faqsFr = [
    { category: 'Livraison', question: 'Comment suivre ma livraison ?', answer: 'Connectez-vous à votre espace Eolis Connect et consultez vos demandes. Votre référence de dossier vous permet de suivre l\'avancement en temps réel.' },
    { category: 'Livraison', question: 'Que faire si mon conteneur est bloqué au port ?', answer: 'Créez une demande dans la catégorie "Livraison > Conteneur bloqué". Notre équipe traitera votre demande en priorité élevée.' },
    { category: 'Livraison', question: 'Quels documents faut-il pour récupérer un conteneur ?', answer: 'Vous aurez besoin de : la facture commerciale, le connaissement (Bill of Lading), la déclaration en douane, et une pièce d\'identité valide.' },
    { category: 'Facturation', question: 'Comment obtenir ma facture ?', answer: 'Vos factures sont disponibles dans votre espace client. Vous pouvez également en faire la demande via le formulaire "Facturation > Demande d\'information".' },
    { category: 'Facturation', question: 'Comment contester une facture ?', answer: 'Créez une demande dans "Facturation > Paiement incomplet" ou "Facturation > Remboursement" selon votre situation. Joignez votre facture et les justificatifs.' },
    { category: 'Dossier', question: 'Quels documents dois-je fournir pour un dossier incomplet ?', answer: 'Les documents requis dépendent du type de prestation. Notre service client vous indiquera exactement ce dont vous avez besoin via la messagerie de votre dossier.' },
    { category: 'Information', question: 'Quels services propose Eolis Cameroun ?', answer: 'Eolis Cameroun propose des services de manutention portuaire, transport de conteneurs (20 et 40 pieds), fret conventionnel, et gestion documentaire.' },
    { category: 'Information', question: 'Quels sont les horaires du service client ?', answer: 'Notre service client est disponible du lundi au vendredi de 8h00 à 18h00. Pour les urgences, un numéro d\'astreinte est disponible.' },
  ]

  for (let i = 0; i < faqsFr.length; i++) {
    await db.fAQ.create({ data: { ...faqsFr[i], locale: 'fr', order: i, subcategory: null } })
  }

  // FAQs EN
  const faqsEn = [
    { category: 'Delivery', question: 'How do I track my delivery?', answer: 'Log in to your Eolis Connect account and check your requests. Your reference number allows you to track the progress in real time.' },
    { category: 'Delivery', question: 'What should I do if my container is blocked at the port?', answer: 'Create a request under "Delivery > Blocked container". Our team will process your request with high priority.' },
    { category: 'Delivery', question: 'What documents are needed to pick up a container?', answer: 'You will need: commercial invoice, Bill of Lading, customs declaration, and a valid ID.' },
    { category: 'Billing', question: 'How do I get my invoice?', answer: 'Your invoices are available in your client area. You can also request one through the form under "Billing > Information request".' },
    { category: 'Billing', question: 'How do I dispute an invoice?', answer: 'Create a request under "Billing > Incomplete payment" or "Billing > Refund" depending on your situation. Attach your invoice and supporting documents.' },
    { category: 'Information', question: 'What services does Eolis Cameroun offer?', answer: 'Eolis Cameroun offers port handling, container transport (20ft and 40ft), conventional freight, and document management services.' },
    { category: 'Information', question: 'What are customer service hours?', answer: 'Our customer service is available Monday to Friday from 8:00 AM to 6:00 PM. For emergencies, an on-call number is available.' },
  ]

  for (let i = 0; i < faqsEn.length; i++) {
    await db.fAQ.create({ data: { ...faqsEn[i], locale: 'en', order: i, subcategory: null } })
  }

  console.log('✅ Database seeded successfully!')
  console.log('')
  console.log('📋 Test accounts (login: username + password):')
  console.log('  System Admin : Christian.DENMEKO / Admin@2026!')
  console.log('  Ops Admin    : Debora.DENMEKO    / Ops@2026!')
  console.log('  Agent 1      : Jean.MBARGA        / Agent@2026!')
  console.log('  Agent 2      : Marie.NGUEMA       / Agent@2026!')
  console.log('  Client 1     : Thomas.KAMGA       / Client@2026!')
  console.log('  Client 2 (EN): Alice.FONO         / Client@2026!')
  console.log('  Pending      : Robert.ETOA        / Client@2026!')
}

main()
  .catch(console.error)
  .finally(() => db.$disconnect())
